### ISSUE TITLE
torch.{tensor, Tensor, LongTensor, ...} isn't captured under `enable_python_mode()`

### ISSUE THREAD

**Issue by zou3519** [pytorch/pytorch#68622]
## 🐛 Bug

See title

## To Reproduce

```
from contextlib import contextmanager
import torch

from torch._C import _DisableTorchDispatch
from torch.utils._python_dispatch import enable_python_mode


n = torch.ones([10, 10])  


@contextmanager
def no_dispatch():
    guard = _DisableTorchDispatch()
    try:
        yield
    finally:
        del guard


class TestTensor(torch.Tensor):
    @classmethod
    def __torch_dispatch__(cls, func, types, args=(), kwargs=None):
        assert False

with enable_python_mode(TestTensor):
    x = torch.tensor(1.)
```
The above script does not raise an error but we would expect all pytorch operations to be caught and sent to the `__torch_dispatch__` under `enable_python_mode`.

## Expected behavior

torch.tensor should dispatch to `__torch_dispatch__` under `enable_python_mode`

## Environment

PT master

**Comment by dagitses** [comment]
I agree this belongs under module: __torch_dispatch and that this is a defect. @albanD , do you concur?

I am not labeling this high-priority because this behavior has likely been this way for a while and it is likely to be noticed easily. Not exactly a silent failure but obvious behavior would not be happening.

Is this blocking anything you're doing right now?

**Comment by albanD** [comment]
It is being tracked in planning. Not high pri indeed.
Some of these will be (very) hard to fix.
